src/main/java/com/hierynomus/spnego/NegTokenInit.java

```diff
@@ -102,11 +103,18 @@ public NegTokenInit read(byte[] bytes) throws IOException {
     public NegTokenInit read(Buffer<?> buffer) throws IOException {
         try {
             ASN1Primitive applicationSpecific = new ASN1InputStream(buffer.asInputStream()).readObject();
-            if (!(applicationSpecific instanceof ASN1ApplicationSpecific)) {
+            if (!(applicationSpecific instanceof BERApplicationSpecific || applicationSpecific instanceof DERApplicationSpecific)) {
```

Comment:
Smbj will continue to use 1.54 and advertise the dependency as 1.54. Just that when someone is using smbj along with some other dependency and understands the risk and wants to forcefully change the boucycastle dependency to version 1.51 in their own builds. If we cant get this, then we wont be able to use the master version for some time.